src/main/java/mat/server/util/XmlProcessor.java

```diff
@@ -42,1198 +42,1167 @@
+    }
+
+    /**
+     * Update node text.
+     *
+     * @param nodeName  the node name
+     * @param nodeValue the node value
+     * @return the string
+     */
+    public String updateNodeText(String nodeName, String nodeValue) {
+        try {
+            LOG.info("In updateNodeText() method");
+            InputSource xmlStream = new InputSource(new StringReader(
+                    originalXml));
+            Document doc = docBuilder.parse(xmlStream);
+            doc.getElementsByTagName(nodeName).item(0).setTextContent(nodeValue);
+            LOG.info("update NoedText");
+            return transform(doc);
+        } catch (Exception e) {
+            LOG.info("Exception thrown on updateNodeText() method");
+            caughtExceptions(e);
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Gets the original xml.
+     *
+     * @return the original xml
+     */
+    public String getOriginalXml() {
+        return originalXml;
+    }
+
+    /**
+     * Sets the original xml.
+     *
+     * @param originalXml the new original xml
+     */
+    public void setOriginalXml(String originalXml) {
+        this.originalXml = originalXml;
+    }
+
+    /**
+     * Gets the xml by tag name.
+     *
+     * @param tagName the tag name
+     * @return the xml by tag name
+     */
+    public String getXmlByTagName(String tagName) {
+        String returnVar = null;
+        Node node = originalDoc.getElementsByTagName(tagName).getLength() > 0 ? originalDoc
+                .getElementsByTagName(tagName).item(0) : null;
+        if (null != node) {
+            returnVar = transform(node);
+        }
+        return returnVar;
+    }
+
+    /**
+     * Adds the parent node.
+     *
+     * @param parentTagName the parent tag name
+     */
+    public void addParentNode(String parentTagName) {
+        if (originalDoc.hasChildNodes()) {
+            Document newDoc = docBuilder.newDocument();
+            Node parentNode = newDoc.appendChild(newDoc
+                    .createElement(parentTagName));
+            Node importedNode = newDoc.importNode(originalDoc.getFirstChild(),
+                    true);
+            parentNode.appendChild(importedNode);
+            originalDoc = newDoc;
+        }
+    }
+
+    /**
+     * Caught exceptions.
+     *
+     * @param excp the exception that was caught
+     */
+    private void caughtExceptions(Exception excp) {
+        if (excp instanceof ParserConfigurationException) {
+            LOG.info("Document Builder Object creation failed"
+                    + excp.getStackTrace());
+        } else if (excp instanceof SAXException) {
+            LOG.info("Xml parsing failed:" + excp.getStackTrace());
+        } else if (excp instanceof IOException) {
+            LOG.info("Conversion of String XML to InputSource failed"
+                    + excp.getStackTrace());
+        } else {
+            LOG.info("Generic Exception: " +
```

Comment:
instead of e.printStackTrace() its better to use LOG.error and pass in the exception. That way you control where the exception gets dumped in the logging configuration.